docs/sources/flow/reference/components/otelcol.receiver.kafka.md

```diff
@@ -0,0 +1,304 @@
+
+Retrieving metadata may fail if the Kafka broker is starting up at the same
+time as the `otelcol.receiver.kafka` component. The [`retry` child
+block][retry] can be provided to customize retry behavior.
+
+### retry block
+
+The `retry` block configures how to retry retrieving metadata when retrieval
+fails.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`max_retries` | `number` | How many times to reattempt retrieving metadata. | `3` | no
+`backoff` | `duration` | Time to wait between retries. | `"250ms"` | no
+
+### autocommit block
+
+The `autocommit` block configures how to automatically commit updated topic
+offsets back to the Kafka brokers.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`enable` | `bool` | Enable autocommitting updated topic offsets. | `true` | no
+`interval` | `duration` | How frequently to autocommit. | `"1s"` | no
+
+### message_marking block
+
+The `message_marking` block configures when Kafka messages are marked as read.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`after_execution` | `bool` | Mark messages after forwarding telemetry data to other components. | `false` | no
+`include_unsuccessful` | `bool` | Whether failed forwards should be marked as read. | `false` | no
+
+By default, a Kafka message is marked as read immediately after it is retrieved
+from the Kafka broker. If the `after_execution` argument is true, messages are
+only read after the telemetry data is forwarded to components specified in [the
+`output` block][output].
+
+When `after_execution` is true, messages are only marked as read when they are
+decoded successfully and components where the data was forwarded did not return
+an error. If the `include_unsuccessful` argument is true, messages are marked
+as read even if decoding or forwarding failed. Setting `include_unsuccessful`
+has no effect if `after_execution` is `false`.
+
+> **WARNING**: Setting `after_execution` to `true` and `include_unsuccessful`
+> to `false` can block the entire Kafka partition if message processing returns
+> a permanent error, such as failing to decode.
+
+### output block
+
+{{< docs/shared lookup="flow/reference/components/output-block.md" source="agent" >}}
+
+## Exported fields
+
+`otelcol.receiver.kafka` does not export any fields.
+
+## Component health
+
+`otelcol.receiver.kafka` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`otelcol.receiver.kafka` does not expose any component-specific debug
+information.
+
+## Example
+
+This example forwards read telemetry data through a batch processor before
+finally sending it to an OTLP-capable endpoint:
+
+```river
+otelcol.receiver.kafka "default" {
+  brokers = ["localhost:9092"]
```

Comment:
Should we also add the `protocol_version` as a required field?

Reply:
Yes, thanks for the catch 👍